src/app/modules/ibc.rs

```diff
@@ -133,44 +184,30 @@ impl<S: ProvableStore> Ibc<S> {
 
 impl<S: Store> ClientReader for Ibc<S> {
```

Comment:
Is there a reason we use `Store` as a bound here (as well as all the other `impl` blocks) as opposed to `ProvableStore`? `S` will always have `ProvableStore` as a supertrait (otherwise we can't even construct the `Ibc` module), therefore I find it better, if not only for documentation purposes, to use `ProvableStore` as a bound everywhere.

WDYT?

Reply:
As discussed during our sync review, we're missing an immutable/versioned private/non-provable store in basecoin-rs -> https://github.com/informalsystems/basecoin-rs/issues/39. So I think it makes sense to specify the trait bound based on what the impl block uses. 👍